### ISSUE TITLE
Stuck on Calculating NFTs unique configuration for *** 

### ISSUE THREAD

**Issue by crlsgzmn** [Solseum/solseum-nft-generator#19]
Each time I try to generate a NFT collection, it gets stuck in Calculating NFTs unique configuration for (Whitelist, etc). 

All the images have the same size. 65,000~ is the maximum possibilities and I'm generating 9,998. 

How can I debug this?

**Comment by Solseum** [comment]
Is because the max to generate takes in count ur background as another layer, the generator ignore that to not have nfts that just differs in the background

**Comment by raymondskl** [comment]
i'm running into the same issue

RANDOMIZE OUTPUT ACTIVE... NFTs LOCATION WILL BE SHUFFLED BEFORE THE METADATA+PNG IS GENERATED.
THE MEANING OF THIS IS: YOUR 0.json THAT SHOULD BE NFT #0, CAN BE THE NFT #1234 AND SO ON.

Cleaning output folder... Done.
Obtainig attributes list and order of layers... Done.
Obtaining Background items and creating the tombola... Done.
Obtaining Body items and creating the tombola... Done.
Obtaining Eyes items and creating the tombola... Done.
Obtaining Head items and creating the tombola... Done.
Obtaining Horn items and creating the tombola... Done.
Obtaining Mouth items and creating the tombola... Done.
Calculating max possibilittes... You can create a max of 7560 NFTs.
[2500, 0, 0] = 2500  NFTs will be created.
Calculating NFTs unique configuration for public_mint_assets ...

**Comment by Solseum** [comment]
You dont have enough combinations. The Max number is using the background as an option, but the generator dont take it as an option

**Comment by raymondskl** [comment]
I've got 6 Background images though

**Comment by Solseum** [comment]
Yeah, but the code ignore the amount of backgrounds to not have the same character with just a different background.

So basically the max amount of combinations is a multiplication between the number of item in each layer but ignoring the background layer.

**Comment by raymondskl** [comment]
Ah I see, that make sense now

**Comment by crlsgzmn** [comment]
> You dont have enough combinations. The Max number is using the background as an option, but the generator dont take it as an option

Hi,

I don't understand what do you mean by a background: 

I have 5 Layers:

1 Background Color 9 items (summing 100%)
2 Vector Type 6 Items (summing 100%)
3 Ace Frame 4 Items
4 Animal Type 19 items (another vector)
5 Key type 16 items.

It fails with -p 4500 -w 4500 -g 2500 -r 1
It only works with less than 5000 and using only one option (for example whitelist), in my case I'm running the process in windows 11 (SF Pro 7 8GB) and in a Rasbarry pi 4+ 8GB. 

C:\Users\drcag\OneDrive\Desktop\NFT final\solseum-nft-generator>python3 main.py -p 4500 -w 2500 -g 2500
Cleaning output folder... Done.
Obtainig attributes list and order of layers... Done.
Obtaining Background Color items and creating the tombola... Done.
Obtaining Vector Type items and creating the tombola... Done.
Obtaining Ace Frame items and creating the tombola... Done.
Obtaining Animal Type Color items and creating the tombola... Done.
Obtaining Key Type Color items and creating the tombola... Done.
Calculating max possibilittes... You can create a max of 65664 NFTs.
[4500, 2500, 2500] = 9500  NFTs will be created.
Calculating NFTs unique configuration for public_mint_assets ... Done.
Created 4500 uniques NFTs for public_mint_assets
Calculating NFTs unique configuration for whitelist_mint_assets ... Done.
Created 2500 uniques NFTs for whitelist_mint_assets
Calculating NFTs unique configuration for giveaway_assets ...

It get's stuck right there. 

It's less than 20% like the readme.md recommends. 

 I really like the program, if you need anything besides a donation just let me know(A translation to spanish for example).

**Comment by Solseum** [comment]
Yeah its a 57k possible combinations counting the 9 assets on background,but the code doesnt take that background to avoid equal nfts with just a different assets so for the code you actual have (6x4x19x16) = 6384 and thats less than ur 10k you wanna build

**Comment by Solseum** [comment]
>> I really like the program, if you need anything besides a donation just let me know(A translation to spanish for example).

Hahaha all donations are welcome my wallet is wrathiontbp.sol

Dw about spanish translation I speak spanish so I can do it!

**Comment by crlsgzmn** [comment]
How do I overcome that? or does it assume that the first layer is always the background?

I want to have to have the background as an asset. 

Que bueno que hablas en español! Saludos!

**Comment by Solseum** [comment]
The background will always appear and will asume that the first layer is the background. But if you wanna have ur expected output you can change one line of code inside nft_creator.py.

Change the line 167 from:
if nftDNA[1:] == nftt[1:]:

To:
if nftDNA[:] == nftt[:]:

Then, you will be able to create the 10k collection using the 57k possible combinations, but you will have a lot of nfts that looks the same (the same attributes and the only one that changes is the background)
